modules/aws/ec2-files.go

```diff
@@ -133,3 +148,102 @@ func FetchContentsOfFilesFromAsgE(t *testing.T, awsRegion string, sshUserName st
+
+	if !files.FileExists(finalLocalDestDir) {
+		os.MkdirAll(finalLocalDestDir, 0755)
+	}
+
+	scpOptions := ssh.ScpDownloadOptions{
+		RemoteHost:      host,
+		RemoteDir:       remoteDirectory,
+		LocalDir:        finalLocalDestDir,
+		FileNameFilters: filenameFilters,
+	}
+
+	return ssh.ScpDirFromE(t, scpOptions, useSudo)
+}
+
+// FetchContentsOfFilesFromAsgE looks up the EC2 Instances in the given ASG, looks up the public IPs of those EC2
+// Instances,, connects to each Instance via SSH using the given username and Key Pair, downloads the files
+// matching filenameFilters at the given remoteDirectory (using sudo if useSudo is true), and stores the files locally
+// at localDirectory/<publicip>/<remoteFolderName>
+func FetchFilesFromAsg(t *testing.T, awsRegion string, sshUserName string, keyPair *Ec2Keypair, asgName string, useSudo bool, remoteDirectory string, localDirectory string, filenameFilters []string) {
+	err := FetchFilesFromAsgE(t, awsRegion, sshUserName, keyPair, asgName, useSudo, remoteDirectory, localDirectory, filenameFilters)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// FetchContentsOfFilesFromAsgE looks up the EC2 Instances in the given ASG, looks up the public IPs of those EC2
+// Instances,, connects to each Instance via SSH using the given username and Key Pair, downloads the files
+// matching filenameFilters at the given remoteDirectory (using sudo if useSudo is true), and stores the files locally
+// at localDirectory/<publicip>/<remoteFolderName>
+func FetchFilesFromAsgE(t *testing.T, awsRegion string, sshUserName string, keyPair *Ec2Keypair, asgName string, useSudo bool, remoteDirectory string, localDirectory string, filenameFilters []string) error {
+	instanceIDs, err := GetInstanceIdsForAsgE(t, asgName, awsRegion)
+	if err != nil {
+		return err
+	}
+
+	errorsOccurred := []error{}
+
+	for _, instanceID := range instanceIDs {
+		err = FetchFilesFromInstanceE(t, awsRegion, sshUserName, keyPair, instanceID, useSudo, remoteDirectory, localDirectory, filenameFilters)
+
+		if err != nil {
+			errorsOccurred = append(errorsOccurred, err)
+		}
+	}
+
+	return customerrors.NewMultiError(errorsOccurred...)
+}
+
+// TODO: Rename this!FetchFilesFromAllAsgsE looks up the EC2 Instances in the given ASG, looks up the public IPs of those EC2
+// Instances,, connects to each Instance via SSH using the given username and Key Pair, downloads the files
+// matching filenameFilters at the given remoteDirectory (using sudo if useSudo is true), and stores the files locally
+// at localDirectory/<publicip>/<remoteFolderName>
+func FetchFilesFromAllAsgsE(t *testing.T, awsRegion string, spec RemoteFileSpecification) error {
+	errorsOccurred := []error{}
+
+	for _, curAsg := range spec.AsgNames {
+		for curRemoteDir, fileFilters := range spec.RemotePathToFileFilter {
+			err := FetchFilesFromAsgE(t, awsRegion, spec.SshUser, spec.KeyPair, curAsg, spec.UseSudo, curRemoteDir, spec.LocalDestinationDir, fileFilters)
```

Comment:
Should I just merge `FetchFilesFromAsgE` into this function and not allow access to it as a separate thing?

Reply:
Yea, a single `FetchFilesFromAsgs` seems good enough

Reply:
Done - consolidated.